internal/core/services/operation_manager/operation_manager.go

```diff
@@ -267,13 +267,22 @@ func (om *OperationManager) StartLeaseRenewGoRoutine(operationCtx context.Contex
 }
 
 func (om *OperationManager) addOperationsLeaseData(ctx context.Context, schedulerName string, ops []*operation.Operation) error {
+	opMap := make(map[string]*operation.Operation)
+	opIds := make([]string, 0, len(ops))
 	for _, op := range ops {
-		ttl, err := om.leaseStorage.FetchLeaseTTL(ctx, schedulerName, op.ID)
-		if err != nil {
-			return fmt.Errorf("failed to fetch lease data for scheduler %s operation %s: %w", op.ID, schedulerName, err)
-		}
-		op.Lease = operation.OperationLease{OperationID: op.ID, Ttl: ttl}
+		opMap[op.ID] = op
+		opIds = append(opIds, op.ID)
+	}
+
+	leases, err := om.leaseStorage.FetchOperationsLease(ctx, schedulerName, opIds...)
+	if err != nil {
+		return fmt.Errorf("failed to fetch operations lease for scheduler %s: %w", schedulerName, err)
+	}
+
+	for _, lease := range leases {
+		opMap[lease.OperationID].Lease = lease
```

Comment:
WDYT about creating a method "setLease()" for the entity operation and use it here? Maybe it would be a good start for future validations

Reply:
https://github.com/topfreegames/maestro/pull/257/commits/f452b6b4816c76362929e2c3e3109afc4fc5c655